garden-backend-service/src/api/schemas/mdf/dataset.py

```diff
@@ -1,12 +1,28 @@
-from pydantic import Field
+from uuid import UUID
 
-from .base import BaseSchema, UniqueList
+from pydantic import Field, RootModel
 
+from ..base import BaseSchema, UniqueList
+from ..search.globus_search import GSearchResult, LegacyResult, ModernResult
 
-class MDFDatasetMetadata(BaseSchema):
-    doi: str
-    versioned_source_id: str
-    source_name: str
-    version: str
 
-    previous_versions: UniqueList[str] = Field(default_factory=list)
+class AccelerateDatasetMetadata(BaseSchema):
+    owner_identity_id: UUID
+    connected_entrypoints: UniqueList[str] = Field(default_factory=list)
+    previous_versions: UniqueList[str] | None = Field(default=None)
+
+
+class MDFModernResult(BaseSchema, ModernResult):
+    accelerate_metadata: AccelerateDatasetMetadata | None = None
+
+
+class MDFLegacyResult(BaseSchema, LegacyResult):
+    accelerate_metadata: AccelerateDatasetMetadata | None = None
+
+
+class MDFGMetaResult(BaseSchema, RootModel[MDFLegacyResult | MDFModernResult]):
```

Comment:
Love these! I think using a RootModel to capture the legacy and modern result schemas together is a nice touch, too